Source/WebGPU/WGSL/Parser.h

```diff
@@ -26,12 +26,65 @@
 #pragma once
 
 #include "AST/ShaderModule.h"
+#include "AST/Statements/ReturnStatement.h"
 #include "CompilationMessage.h"
 #include "Lexer.h"
 #include <wtf/Expected.h>
 
 namespace WGSL {
 
```

Comment:
Can we put this in a ParserPrivate.h or ParserTesting.cpp instead? It’s important to be able to distinguish between real API and testing API.

Reply:
I'm also thinking of something like that but `ParserImpl.h/cpp`. In that case do we have to set the target membership of `ParserImpl.h` then?

Also one thing, right now I have a *lot* of changes in `Parser.cpp`. If I move everything to `ParserImpl.cpp` Git will freak out about the changes + rename. Maybe we'll actually move it after we're done with upstreaming all changes?